apps/webapp-registry/setup.sh

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env bash
+
+######
+# setup.sh
+#
+# Sets up infrastructure for a continuously-built and deployed container-based
+# web app for Azure App Service.
+#
+# $1: image_name: Name of image (aka repo) to use or create. Defaults to
+#     $IMAGE_NAME.
+# $2: image_tag: Name of image tag. Defaults to $IMAGE_TAG
+# $3: repo_url: Source code repo to use for continuous build. Defaults to
+#     "https://github.com/{IMAGE_NAME}.git"
+# $4: base_name: A default prefix for all Azure resources. Defaults to $AZURE_BASE_NAME.
+# $5: registry_name: Name of Azure container registry to use or create.
+#     Defaults to "${AZURE_BASE_NAME}-registry".
+# $6: app_name: Name of App Service web app to use or create. Defaults to
+#     "{AZURE_BASE_NAME}-webapp".
+# $7: plan_name: Name of App Service plan to use or create. Defaults to
+#     "{AZURE_BASE_NAME}-plan".
+# $8: group_name: Name of Azure resource group to use for all resources.
+#     Defaults to "${AZURE_BASE_NAME-group".
+# $9: location: Name of Azure location to use for all resources. Defaults to
+#     $AZURE_DEFAULT_LOCATION.
+#####
+
+__filename=${BASH_SOURCE[0]}
+__dirname=$(cd $(dirname ${__filename}) && pwd)
+__root=${__dirname}
+if [[ ! -f "${__root}/.env" ]]; then cp "${__root}/.env.tpl" "${__root}/.env"; fi
+if [[ -f "${__root}/.env" ]]; then source "${__root}/.env"; fi
+source "${__dirname}/scripts/rm_helpers.sh"  # for ensure_group
+
+image_name=${1:-${IMAGE_NAME}}
+image_tag=${2:-${IMAGE_TAG}}
+repo_url=${3:-"https://github.com/${image_name}.git"}
+
+base_name=${4:-"${AZURE_BASE_NAME}"}
+registry_name=${5:-"$(echo ${base_name} | sed 's/[- _]//g')registry"}
+app_name=${6:-"${base_name}-webapp"}
+plan_name=${7:-"${base_name}-plan"}
+group_name=${8:-"${base_name}-group"}
+location=${9:-${AZURE_DEFAULT_LOCATION}}
+
+# set after getting registry config
+image_uri=
+registry_sku=Standard
+url_suffix=azurewebsites.net
+
+# errors
+declare -i err_registrynameexists=101
+declare -i err_nogithubtoken=102
+
+####
+
+## ensure groups
+ensure_group $group_name
+
+## ensure registry
+registry_id=$(az acr show \
```

Comment:
At some point here during my initial test, the script hung, but I can't see what would have caused it. There's the possibility that it was caused by Docker not running locally, but would need to investigate further. I can say for sure that the resource group was created, but had no resources allocated to it.